### DISCUSSION TITLE
Wrong date is displayed

### DISCUSSION THREAD

**Discussion by mvorisek** [community/community#35175] [Repositories]
This problem took me quite a while to figure out, currently the UI displays a wrong date which is very misleading.

Steps to reproduce:
1. open https://github.com/atk4/ui/blob/d7bdc09dceba2e8575ff0ae39e9ee12ab60e9781/public/atkjs-ui.js
2. notice `Apr 15` date is displayed:
   ![image](https://user-images.githubusercontent.com/2228672/194012684-8485b872-8abd-4d74-b82c-5c18d7b4faba.png)
3. but the date should be `Aug 28`, the same date as the commit - https://github.com/atk4/ui/commit/d7bdc09dceba2e8575ff0ae39e9ee12ab60e9781

I belive this is a bug. If not, please explain me where does the the `Aug 15` date comes from.

**Comment by airtower-luna** [comment]
`Apr 15` (not Aug!) seems to be the date listed in the commit itself as author date: https://github.com/atk4/ui/commit/d7bdc09dceba2e8575ff0ae39e9ee12ab60e9781.patch

Has the commit been rebased in August (including as part of a rebase-merge)? That'd explain why author and committer date are different.

**Reply by mvorisek** [reply]
thanks for the reply - rebased and amended

it is still quite confusing as the commit history - https://github.com/atk4/ui/commits/d7bdc09dceba2e8575ff0ae39e9ee12ab60e9781/public/atkjs-ui.js - has a different (and expected by me) date

maybe this can be rethink or is there any reason for the 2 different dates in the same/GH ecosystem?

**Reply by airtower-luna** [reply]
Git stores two dates for each commit: The *author date* and the *committer date*. When a commit is initially created they are the same. If the commit is recreated later (e.g. in a rebase, or from a mailed patch) only the committer date is updated, and whoever does the recreation set as the committer while the author information stays the same. The point is that both the original author and whoever applied the commit remain visible. `git log --pretty=fuller` will show you both author and committer info.

So much for Git in general. I don't know either why GitHub made the choice to show the committer date in most places and author date in others. The "committed on" is only really telling if you know what to look for. :sweat_smile:
